trieste/acquisition/function.py

```diff
@@ -746,6 +746,57 @@ def ehvi_based_on_partitioned_cell(
     return acquisition
 
 
+class ExpectedConstrainedHypervolumeImprovement(ExpectedConstrainedImprovement):
+    """
+    Builder for the constrained expected hypervolume improvement acquisition function.
+    This function essentially combines ExpectedConstrainedImprovement and ExpectedHypervolumeImprovement.
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return (
+            f"ExpectedConstrainedHypervolumeImprovement({self._objective_tag!r}, {self._constraint_builder!r},"
+            f" {self._min_feasibility_probability!r})"
+        )
+
+    def prepare_acquisition_function(
+        self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
+    ) -> AcquisitionFunction:
```

Comment:
this function is a combination of eci and ehvi, and should be possible to nicely abstract so that you can switch out ei vs ehvi easily